### PR TITLE
fix lifecycleOwner for java

### PR DESCRIPTION
**Pull request by hannesa2**
I found more places which prevent an error free `./gradlew check` It's a follow up from #71

Now it builds properly, you can see it with [Github actions](https://github.com/hannesa2/mlkit/pull/1/checks?check_run_id=884639015)

### CODE REVIEW

File: android/translate/app/src/main/java/com/google/mlkit/samples/translate/java/TranslateFragment.java [hunk 1]
```diff
@@ -76,8 +76,7 @@ public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceStat
         final Spinner sourceLangSelector = view.findViewById(R.id.sourceLangSelector);
         final Spinner targetLangSelector = view.findViewById(R.id.targetLangSelector);
 
-        final TranslateViewModel viewModel =
-                ViewModelProviders.of(this).get(TranslateViewModel.class);
+        final TranslateViewModel viewModel = ViewModelProviders.of(this).get(TranslateViewModel.class);
```
Comment (rosariopfernandes) [inline android/translate/app/src/main/java/com/google/mlkit/samples/translate/java/TranslateFragment.java:79]: Unrelated to this PR, but `ViewModelProviders` was deprecated in favor of [ViewModelProvider](https://developer.android.com/reference/androidx/lifecycle/ViewModelProvider).

Comment (hannesa2) [reply android/translate/app/src/main/java/com/google/mlkit/samples/translate/java/TranslateFragment.java:79]: There are **a lot** of other things to improve.
I see a better chance to get merged by doing it step by step

File: android/smartreply/app/src/main/java/com/google/mlkit/samples/smartreply/java/chat/ChatFragment.java [hunk 2]
```diff
@@ -138,14 +138,14 @@ public void onClick(View view) {
...
-        mViewModel.getSuggestions().observe(this, new Observer<List<SmartReplySuggestion>>() {
+        mViewModel.getSuggestions().observe(getViewLifecycleOwner(), new Observer<List<SmartReplySuggestion>>() {
             @Override
             public void onChanged(List<SmartReplySuggestion> suggestions) {
                mChipAdapter.setSuggestions(suggestions);
             }
         });
 
-        mViewModel.getMessages().observe(this, new Observer<List<Message>>() {
+        mViewModel.getMessages().observe(getViewLifecycleOwner(), new Observer<List<Message>>() {
```
Comment (Kagdxdd) [inline android/smartreply/app/src/main/java/com/google/mlkit/samples/smartreply/java/chat/ChatFragment.java:148]: - [ ]

Comment (Kagdxdd) [reply android/smartreply/app/src/main/java/com/google/mlkit/samples/smartreply/java/chat/ChatFragment.java:148]: 79903e6a756d3d7473b6aefc5b34829b662e57a3

File: android/smartreply/app/src/main/java/com/google/mlkit/samples/smartreply/java/chat/ChatFragment.java [hunk 3]
```diff
@@ -155,7 +155,7 @@ public void onChanged(List<Message> messages) {
             }
         });
 
-        mViewModel.getEmulatingRemoteUser().observe(this, new Observer<Boolean>() {
+        mViewModel.getEmulatingRemoteUser().observe(getViewLifecycleOwner(), new Observer<Boolean>() {
```
Comment (Kagdxdd) [inline android/smartreply/app/src/main/java/com/google/mlkit/samples/smartreply/java/chat/ChatFragment.java:158]: - [ ]

**Review by Kagdxdd — APPROVED** [review]

**Comment by Kagdxdd** [comment]
- [ ]

**Pull request merged**